pkg/client/unversioned/remotecommand/remotecommand.go

```diff
@@ -195,49 +195,37 @@ func (e *Streamer) doStream() error {
 			return err
 		}
 		defer remoteStdin.Reset()
-		// TODO this goroutine will never exit cleanly (the io.Copy never unblocks)
-		// because stdin is not closed until the process exits. If we try to call
-		// stdin.Close(), it returns no error but doesn't unblock the copy. It will
-		// exit when the process exits, instead.
+		wg.Add(1)
 		go cp(api.StreamTypeStdin, remoteStdin, e.stdin)
 	}
 
-	waitCount := 0
-	completedStreams := 0
-
 	if e.stdout != nil {
-		waitCount++
 		headers.Set(api.StreamType, api.StreamTypeStdout)
 		remoteStdout, err := conn.CreateStream(headers)
 		if err != nil {
 			return err
 		}
 		defer remoteStdout.Reset()
+		wg.Add(1)
 		go cp(api.StreamTypeStdout, e.stdout, remoteStdout)
 	}
 
 	if e.stderr != nil && !e.tty {
-		waitCount++
 		headers.Set(api.StreamType, api.StreamTypeStderr)
 		remoteStderr, err := conn.CreateStream(headers)
 		if err != nil {
 			return err
 		}
 		defer remoteStderr.Reset()
+		wg.Add(1)
 		go cp(api.StreamTypeStderr, e.stderr, remoteStderr)
 	}
 
-Loop:
-	for {
-		select {
-		case <-doneChan:
-			completedStreams++
-			if completedStreams == waitCount {
-				break Loop
-			}
-		case err := <-errorChan:
-			return err
-		}
+	wg.Wait()
```

Comment:
waiting here means a hung stdout/stderr stream can keep us from returning, even if we encountered an error